src/Codeception/Lib/Connector/ZF2.php

```diff
@@ -163,15 +163,12 @@ public function addServiceToContainer($name, $service)
 
     private function createApplication()
     {
-        $this->application = Application::init($this->applicationConfig);
+        $this->application = Application::init(array_merge($this->applicationConfig, [
+            'service_manager' => [
+                'services' => $this->persistentServices
+            ]
+        ]));
```

Comment:
You are using `array_merge` to merge two-dimensional array,
I think that it could cause issues if you had some options for `service_manager` set in applicationConfig.

Probably it would be better to set specific key in array:
```php
$config = $this->applicationConfig;
$config['service_manager']['services'] = $this->persistentServices;
$this->application = Application::init($config);
```

`array_merge_recursive` is an option too.

Reply:
I have asked ZF guys and they said just use `ArrayUtils::merge`

> developed specifically for config merging

I will update